mqtt/policy/src/core/builder.rs

```diff
@@ -687,17 +706,17 @@ mod tests {
             .with_default_decision(Decision::Denied)
             .build();
 
-        assert_matches!(result, Err(Error::ValidationSummary(errors)) if errors.len() == 1 );
+        assert_matches!(result, Err(Error::Validation(_)));
     }
 
     #[derive(Debug)]
     struct FailAllValidator;
 
     impl PolicyValidator for FailAllValidator {
-        fn validate(&self, _definition: &PolicyDefinition) -> Result<()> {
-            Err(Error::ValidationSummary(vec![Error::Validation(
-                "Unable to parse policy definition".to_string(),
-            )]))
+        type Error = ValidatorError;
+
+        fn validate(&self, _definition: &PolicyDefinition) -> StdResult<(), Self::Error> {
```

Comment:
will it complain if you put just Result<(), Self::Error> ?

Reply:
Yeah, b/c local Result type is imported from super::*.